src/framework/theme/components/calendar-kit/services/native-date.service.ts

```diff
@@ -125,7 +125,9 @@ export class NbNativeDateService extends NbDateService<Date> {
    * We haven't got capability to parse date using formatting without third party libraries.
    * */
   parse(date: string, format: string): Date {
-    return new Date(Date.parse(date));
+    const parsedValue = new Date(Date.parse(date));
+    const isProperlyParsed = parsedValue && parsedValue.toString() !== 'Invalid Date';
+    return isProperlyParsed ? parsedValue : null;
```

Comment:
It's a breaking change. Users could rely on `parse` always returning a `Date` object. With updated behavior, there is possibly a `null.someDateMember` call. What do you think about checking for invalid dates in the validators themselves?
To check for invalid date, let's better use `isNaN(date.getTime)`. We could add the `isValid` method to the `NbDateService`. Don't forget to update moment implementation (NbMomentDateService) and this method (https://github.com/akveo/nebular/blob/7753027cb9e1efec28cc26a54b18a53b5f2e1eb0/src/framework/theme/components/calendar-kit/services/native-date.service.ts#L68).

Also, I'm concerned about the correctness of min/max validators for range pickers.
https://github.com/akveo/nebular/blob/7753027cb9e1efec28cc26a54b18a53b5f2e1eb0/src/framework/theme/components/datepicker/datepicker.directive.ts#L394
https://github.com/akveo/nebular/blob/7753027cb9e1efec28cc26a54b18a53b5f2e1eb0/src/framework/theme/components/datepicker/datepicker.directive.ts#L404
`datepickerAdapter.parse` on range picker would return a `{ start: D, end: D }` objects, and we use it directly in the `dateService.compareDates` which expects Date objects. Could you please check does it works as expected?

And the last thing, let's change ternary operators in the validators to `if` to make it readable.